File: java-concert/src/main/java/org/example/javaconcert/concert/presentation/request/ConcertRequest.java
package org.example.javaconcert.concert.presentation.request;

import java.time.LocalDate;

import org.example.javaconcert.concert.domain.Concert;
import org.example.javaconcert.concert.domain.Tag;

public record ConcertRequest(
	String title,
	String location,
	LocalDate startAt,
	LocalDate endAt,
	Tag tag
) {

	public Concert toDomain() {
		return Concert.builder()
			.title(title)
			.location(location)
			.startAt(startAt)
			.endAt(endAt)
			.tag(tag)
			.build();
	}
}

File: concert/src/main/kotlin/prac/concert/concert/presentation/request/ConcertRequest.kt
package prac.concert.concert.presentation.request

import prac.concert.concert.domain.Concert
import prac.concert.concert.domain.Tag
import java.time.LocalDate

data class ConcertRequest(
    val title: String,
    val location: String,
    val startAt: LocalDate,
    val endAt: LocalDate,
    val tag: Tag
) {

    fun toDomain(): Concert {
        return Concert(
            title = this.title,
            location = this.location,
            startAt = this.startAt,
            endAt = this.endAt,
            tag = this.tag
        )
    }
}
